src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentHighlighting/DocumentHighlightEndpoint.cs

```diff
@@ -57,18 +57,18 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
 
     protected override async Task<DocumentHighlight[]?> HandleDelegatedResponseAsync(DocumentHighlight[]? response, DocumentHighlightParams request, RazorRequestContext requestContext, DocumentPositionInfo positionInfo, CancellationToken cancellationToken)
     {
-        var documentContext = requestContext.GetRequiredDocumentContext();
-        var codeDocument = await documentContext.GetCodeDocumentAsync(cancellationToken).ConfigureAwait(false);
-
         if (response is null)
         {
             return null;
         }
 
-        var csharpDocument = codeDocument.GetCSharpDocument();
```

Comment:
The alternative fix here would have been to check `positionInfo.LanguageKind` and only map if it was CSharp, but this is more future proof. Sadly its basically untestable too, with our current infra.

Reply:
I hate to say it but... integration test?

Reply:
oh good call, i should have thought of that! Looks like this would be the first highlight test though, so might leave it for a follow up, to work out how to actually get highlights from VS. If there is something I can quickly rip off from Roslyn though, will do it here.

Reply:
I think you end up getting `VisualStudio.Editor.GetTagSpans` for whatever they tag it with....

Reply:
Thanks for the hint, you were spot on, and I never would have got there 😛